jupyter/nbgrader/unit-tests/runUnitTests.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+NBGRADER_LEARNER="learner"
+ASSIGNMENT_NAME="unit-tests"
+
+if [[ "$#" -ne 1 ]]; then
+    echo "Please enter exactly one argument"
+    exit
+fi
+
+NOTEBOOK_FILENAME="$1_tests.ipynb"
+DECODED_FEEDBACK="$1_tests.html"
+
+cd unit-test-env
+
+# # Get kernel language from the submitted assignment's metadata
+kernel_language="$(sed -n 's/"language": //p' "submitted/$NBGRADER_LEARNER/$ASSIGNMENT_NAME/$NOTEBOOK_FILENAME")"
+
+echo "running tests for $1 kernel"
+
+# Scrub given feedback file to remove hidden tests and tracebacks
+# Checks options.json for toggling on/off hiding tests or tracebacks
+python scrub.py "feedback/$NBGRADER_LEARNER/$ASSIGNMENT_NAME/$DECODED_FEEDBACK" 100 "$kernel_language"
```

Comment:
I see `scrub.py` is included in `.gitignore`. Is it intended?

Reply:
Yeah. `Scrub.py` is kept secret to help preserve some secrecy about how we redact feedback. I was hoping that by adding the file to `.gitignore` that git would stop tracking changes made to the file.

Reply:
Hi (from one of the partner universities); if the particular lab/grader can't be rebuilt from source without the missing pieces, maybe there's no need to include it in the public repo. To some extent any attempt to harden nbgrader will rely on obscurity because it doesn't natively support the separation of student and grading code. (I privately reported some other ways that the solution can be dumped without the scrub script catching it.) So I think it does make sense to have some obscurity mechanics remain secret but there's a broader issue. That's just my opinion.

Reply:
@fng-coursera -- I added `scrub.py` to the `.gitignore` file so that we could have a fully functional `scrub.py` file locally, and not have to worry about accidentally including it in a commit. 

@echuber2 -- Thanks for the comments! The nbgrader images can still be built with our stripped-down version of `scrub.py`. It's just that running the autograder won't apply any of our traceback redaction. These images are here so that partners can have something to build off of, and even without any traceback redaction, it's still useful to have the jupyter notebook base here.

I do wish there were a more robust, native way of working with nbgrader to remove sensitive tests/feedback. Do you know who might have those private reports of how a student might bypass the scrub script? I'd love to take a look to see how we might be able to prevent that.

Reply:
@josephl310 Sure, I had a conversation two weeks ago via email/Zoom with Coursera's Alex Plunkett and Kevin Behan about this.